### PR TITLE
contrib: Replace developer keys with list of pgp fingerprints

### PR DESCRIPTION
**Pull request by maflcko**
Having to host a copy of the keys in this repo was a common source of discussion and distraction, caused by problems such as:

* Outdated keys. Unclear whether and when to replace by fresh copies.
* Unclear when to add a key of a new developer or Gitian builder.

The problems are solved by
* Having no keys but only the fingerprints
* Adding a rule of thumb, when to add a new key

<strike>Moving the keys to a different repo solves none of these issues, but since the keys are not bound to releases or git branches of Bitcoin Core, they should live somewhere else.

Obviously, all keys are hosted and distributed on key servers, but were added to the repo solely for convenience and redundancy.

Moving the mirror of those keys to a different repo makes it less distracting to update them -- let's say -- prior to every major release.

I updated our `doc/release-process.md` to reflect the new location.

DEPENDS_ON https://github.com/bitcoin-core/gitian.sigs/pull/621
</strike>

### CODE REVIEW

**Comment by jonasschnelli** [comment]
Moving the keys away from this repository would loose the potential for a "verification-chain" (signed builds contains keys to verify future releaes).
Shipping the keys in a verifiable binary (or now via the tar ball) adds additional security.

My long term use case is to have a binary verifier shipped with bitcoin core (same signing and verification process).
Therefore we could add keys (probably new secp256k1 keys) to the binary and allow verification of future releases.

Maybe we could do that once we have that use case more sketched out.

**Comment by MeshCollider** [comment]
@laanwj suggested leaving the key fingerprints in the main repo (https://github.com/bitcoin/bitcoin/pull/11845#issuecomment-351387138)

**Comment by laanwj** [comment]
Concept ACK.

Having the keys in a separate repository also means that we can keep the keys up to date there - no problem with commit noise, no more need for pulling from gpg keyservers at the beginning of the travis build.

> @laanwj suggested leaving the key fingerprints in the main repo

Yes, having the key fingerprints in the main repo is enough to verify the keys are correct.

File: contrib/gitian-keys/README.md
Comment (laanwj) [inline contrib/gitian-keys/README.md]: Maybe add a separate script? Such a large snippet is hard to copy/paste.

**Comment by luke-jr** [comment]
Would be nice to have names with the fingerprints.

Maybe:

```
...
AEC1884398647C47413C1C3FB1179EB7347DC10D John Doe
" | while read fingerprint keyholder_name; do
...
```

**Comment by MeshCollider** [comment]
Concept ACK

**Comment by jonasschnelli** [comment]
> Having the keys in a separate repository also means that we can keep the keys up to date there - no problem with commit noise, no more need for refreshing keys from gpg keyservers at the beginning of the travis build (which fails half the time).

Good point.

Concept ACK

File: contrib/gitian-keys/README.md
Comment (achow101) [inline contrib/gitian-keys/README.md]: s/builers/builders

**Review by achow101 — COMMENTED** [review]
utACK fa38d529a75a6cd1a51326c05170b3db2606fd14 modulo typo

Maybe this should include the fingerprint of the release key too?

It might also be better to move this to a more generic pgp keys folder instead of specifically `gitian-keys`.

**Comment by MarcoFalke** [comment]
Fixed @achow101 nit and fixed up, should be easy to re-ACK.

> Maybe this should include the fingerprint of the release key too?

Happy to do that, but I think this is separate from the current goal. Will add a commit on top if others agree on that.

> It might also be better to move this to a more generic pgp keys folder instead of specifically gitian-keys.

Note that the fingerprints of the maintenance keys are listed in https://github.com/bitcoin/bitcoin/blob/62fdf9b07087b80d2142799bdd2324f61483359d/contrib/verify-commits/trusted-keys and the ones in this folder are in fact only used for gitian signatures. So I guess the name is fine to keep for now.

File: contrib/gitian-keys/README.md
Comment (fanquake) [inline contrib/gitian-keys/README.md]: nit: "keys.txt contains"

File: contrib/gitian-keys/keys.txt
Comment (fanquake) [inline contrib/gitian-keys/keys.txt]: This key "D2D1085B9425F9DEFA06E7932270E30C522739F6" has expired, and I couldn't look it up. We could probably remove it given @jonasschnelli has another key listed.

File: contrib/gitian-keys/keys.txt
Comment (fanquake) [inline contrib/gitian-keys/keys.txt]: I verified that I was able to retrieve all the other keys, and that the names listed here were correct.

**Comment by practicalswift** [comment]
Concept ACK

File: contrib/gitian-keys/keys.txt
Comment (akx20000a) [inline contrib/gitian-keys/keys.txt]: Could you add my key fingerprints?
617C90010B3BD370B0AC7D424BB42E31C79111B8 Akira Takizawa

**Review by ghost — CHANGES_REQUESTED** [review]

File: contrib/gitian-keys/keys.txt
Comment (MarcoFalke) [reply contrib/gitian-keys/keys.txt]: Removal and addition of keys will happen in separate pull requests. You should wait until this one is merged.  
Might want to help by review, though.

**Comment by laanwj** [comment]
utACK fabb72b

File: contrib/gitian-keys/keys.txt
Comment (laanwj) [reply contrib/gitian-keys/keys.txt]: @akx20000a This has been merged, you can submit your PR now

**Pull request merged**
